modules/rhcos-setting-grub-password-day-2.adoc

```diff
@@ -0,0 +1,144 @@
+// Module included in the follow assemblies:
+//
+// * post_installation_configuration/machine-configuration-tasks.adoc
+
+:_content-type: PROCEDURE
+[id="rhcos-setting-grub-password-day-2_{context}"]
+= Setting a GRUB password on {op-system}
+
+:FeatureName: GRUB password management
+include::snippets/technology-preview.adoc[]
+
+You can set up a password to prevent unauthorized users from accessing the GRUB command line, modify kernel command-line arguments, or boot non-default OSTree deployments.
+
+.Prerequisites
+
+* You have a running {product-title} cluster.
+* You are logged in as a cluster administrator.
+* You installed the OpenShift CLI (`oc`).
+* You installed the `butane` utility.
+* You installed the `grub2-tools-minimal` package.
+
+.Procedure
+
+. Create a password hash for GRUB using the `grub2-mkpasswd-pbkdf2` command line tool:
++
+[source,terminal]
+----
+$ grub2-mkpasswd-pbkdf2
+Enter password:
+Reenter password:
+PBKDF2 hash of your password is
+grub.pbkdf2.sha512.10000.5AE6255...
+----
+
+. Create a script that validates your configurations and sets the GRUB config for the root user with the newly created password:
++
+.rhcos_set_grub_password.sh
+[source,bash]
+----
+set -euo pipefail
+
+STAMPFILE="/var/lib/.grub_password_<date_changed>"
+GRUBPASSWORD="grub.pbkdf2.sha512.10000.58E6255..."
+
+main () {
+
+# Validate GRUB config
+  if [[ $(grep -c 'source $prefix/user.cfg' '/boot/grub2/grub.cfg') != 1]]; then
+    echo "Invalid GRUB config: Missing user.cfg"
+    return 1
+  fi
+
+# Validate ostree repo config
+  if [[ $(grep -c 'bls-append-except-default=grub_users=""' '/sysroot/ostree/repo/config') != 1 ]]; then
+    echo "Invalid ostree repo config: Missing bls-append-except-default"
+    return 1
+  fi
+
+# Remount /boot as RW
+  mount -o remount,rw /boot
+
+# Write config
+  echo -e "set superusers="root"\npassword_pbkdf2 root {GRUBPASSWORD}" > /boot/grub2/user.cfg
+  chmod 400 /boot/grub2/user.cfg
+
+# Write stamp file
+  touch "{STAMPFILE}"
+
+}
+----
+
+. Create a systemd service unit that launches the script on boot:
++
+.rhcos-set-grub-password.service
+[source,text]
+----
+[Unit]
+Description=Set GRUB password
+ConditionPathExists=!/var/lib/.grub_password_<date_changed>
+RequiresMountsFor=/boot
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+MountFlags=slave
+ExecStart=/usr/local/bin/rhcos_set_grub_password.sh
+
+[Install]
+WantedBy=multi-user.target
+----
+
+. Create a Butane config that includes the files from the previous steps:
++
+.40-worker-custom-journald.bu
+[source,yaml]
+----
+variant: openshift
+version: {product-version}.0
```

Comment:
suggest to use 4.13.0, it will generate machine config with ignition ver. 3.2.0. rendered config version is 3.4.0  [4.14.0-experimental](https://coreos.github.io/butane/config-openshift-v4_14-exp/) it not recommended, because it will generate machine config with ignition ver `3.5.0-experimental` which is not supported in 4.14

Reply:
We'll fix that soon and downgrade 4.14 to 3.4: https://github.com/coreos/butane/issues/488
This should stay the on 4.14.0 variant.